skpro/regression/linear/_glm.py

```diff
@@ -167,8 +179,44 @@ class GLMRegressor(BaseProbaRegressor):
         "y_inner_mtype": "pd_DataFrame_Table",
     }
 
+    def _str_to_sm_family(self, family, link):
+        """Convert the string to a statsmodel object.
+
+        If the link function is also explcitly mentioned then include then
+        that must be passed to the family/distribution object.
+        """
+        from warnings import warn
+
+        from statsmodels.genmod.families.family import Gamma, Gaussian, Poisson
+        from statsmodels.genmod.families.links import Identity, InversePower, Log, Sqrt
+
+        sm_fmly = {
+            "Normal": Gaussian,
+            "Poisson": Poisson,
+            "Gamma": Gamma,
+        }
+
+        links = {
+            "Log": Log,
+            "Identity": Identity,
+            "InversePower": InversePower,
+            "Sqrt": Sqrt,
+        }
+
+        if link in links:
+            link_function = links[link]()
+            try:
+                return sm_fmly[family](link_function)
+            except Exception:
+                msg = "Invalid link for family, default link will be used"
+                warn(msg)
+
+        return sm_fmly[family]()
+
     def __init__(
         self,
+        family="Normal",
```

Comment:
it would have been great to have these as starting params from the get-go, but right now we cannot add these as starting parameters due to the deprecation policy - it would make user code break.

We have to add the new params at the end, and I would suggest following deprecation policy to move them to the start eventually, see https://www.sktime.net/en/latest/developer_guide/deprecation.html